plugin/src/generators/lib-tags/generator.ts

```diff
@@ -0,0 +1,198 @@
+import { readJson, Tree, writeJson, getProjects, ProjectConfiguration, formatFiles } from '@nx/devkit';
+import { LibTagsGeneratorSchema } from './schema';
+import { execSync } from 'child_process';
+import { addNxAppTag, addNxScopeTag, askQuestion, getNxRules, getNxRulesEntry, readESLintConfig } from '../../shared/utils';
+import { isEmpty } from 'lodash';
+
+export async function libTagsGenerator(
+  tree: Tree,
+  options: LibTagsGeneratorSchema,
+) {
+  const hasUnstagedChanges = execSync('git status -s').toString('utf8');
+
+  if (hasUnstagedChanges) {
+    const shouldContinue = await askQuestion('You have unstaged changes. Are you sure you want to continue? (y/n): ') === 'y';
+
+    if (!shouldContinue) {
+      return;
+    }
+  }
+
+  const log = (message: string): void => {
+    if (!options.silent) {
+      console.log(message);
+    }
+  };
+
+  // #1 Check eslint config nx-boundaries rule
+  log('1. Checking eslint config nx-boundaries rule...\n');
+  const { config, path } = readESLintConfig(tree);
+
+  if (!config || isEmpty(config)) {
+    throw new Error(`Failed to load ESLint config: ${path}`);
+  }
+
+  try {
+    const rulesEntry = getNxRulesEntry(config).rules['@nx/enforce-module-boundaries'];
+
+    if (rulesEntry[0] !== 'error') {
+      rulesEntry[0] = 'error';
+    }
+
+    if (rulesEntry[1].depConstraints.find((rule) => rule.sourceTag === '*' && rule.onlyDependOnLibsWithTags.includes('*'))) {
+      const esLintConfigTemplate = readJson(tree, 'plugin/src/generators/code-checks/files/.eslintrc.json.template');
+      const templateRules = getNxRulesEntry(esLintConfigTemplate);
+  
+      rulesEntry[1] = templateRules.rules['@nx/enforce-module-boundaries'][1];
+
+      writeJson(tree, path, config);
+    }
+    // TODO: use custom errors
+  } catch {
+    log('ESLint config has no @nx/enforce-module-boundaries rule. Updating rules...\n');
+
+    const esLintConfigTemplate = readJson(tree, 'plugin/src/generators/code-checks/files/.eslintrc.json.template');
+    const templateRules = getNxRulesEntry(esLintConfigTemplate);
+
+    config.overrides.push(templateRules);
+
+    writeJson(tree, path, config);
+  }
+
+  // #2 Check projects tags
+  log('2. Checking projects tags...\n');
+  const projects = getProjects(tree);
+  const applications: Array<ProjectConfiguration> = [];
+  const libraries: Array<ProjectConfiguration> = [];
+  const rules = getNxRules(config);
+
+  const verifyAppTag = (project: ProjectConfiguration, appTag: string): void => {
```

Comment:
If possible, move nested functions outside the generator, please. It will simplify the code perception

Reply:
These functions are part of the generator implementation. I moved `verifyEsLintConfig` to `config-utils`.

Reply:
These functions can be in the same file, but not inside generator

```ts
type TagType = 'app' | 'scope' | 'type';

const getTagFromLibPath = ...

const verifyLibraryTag = ...

const checkApplicationTags = ...

const checkLibraryTags = ...

export async function libTagsGenerator...
```

Reply:
Updated